.circleci/config.yml

```diff
@@ -14,6 +15,26 @@ commands:
           fingerprints:
             - "0e:50:82:ad:81:17:a5:31:04:13:fd:6e:c7:72:c9:18"
 
+  detect_root_change:
+    description: Detects root package.json changes
+    steps:
+      - run: |
+          changes=`git diff --name-only master...$CIRCLE_BRANCH | grep -v '^packages/'`
+
+          if [ "${#changes}" > 0 ];
```

Comment:
if the length of `$CHANGES` > 0, set `ROOT_CHANGE=true` 



This is then used to determine if all packages should be built / listed / tested, or only changed packages.